keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -889,50 +906,42 @@ fields of API types, flags, etc.?**
 _This section must be completed when targeting beta graduation to a release._
 
 * **How can an operator determine if the feature is in use by workloads?**
-  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-  checking if there are objects with field X set) may be a last resort. Avoid
-  logs or events for this purpose.
+
+  Kubelet metrics will be updated to report the number of HostProcess containers started and number of errors started.
+
+  TBD: Confirm the best way to accomplish this is to add new [values/metric labels](https://github.com/kubernetes/kubernetes/blob/fe099b2abdb023b21a17cd6a127e381b846c1a1f/pkg/kubelet/metrics/metrics.go#L96-L99) for `StartedContainersTotal` and `StartedContainersError` counters. Otherwise we could add new counters.
+
+
 
 * **What are the SLIs (Service Level Indicators) an operator can use to determine 
 the health of the service?**
-  - [ ] Metrics
-    - Metric name:
+  - [x] Metrics
+    - Metric name: Add labels to report counts of `HostProcess` containers (host_process_container, host_process_init_container, and host_process_ephemeral_container) to `started_containers_total` and `started_containers_errors_total`
```

Comment:
@ehashman what are your thoughts on adding new labels to `started_containers_total` and `started_containers_errors_total` to identify Windows HostProcesss containers usage?

Reply:
This approach looks good to me if sig-instrumentation says the label is reasonable.

Reply:
Today there are labels for [container, init_container, and ephemerical_container](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/metrics/metrics.go#L96-L100) which are used to track the [cumulative number of containers started](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/metrics/metrics.go#L473) and [cumulative number of containers failing to start](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/metrics/metrics.go#L482-L491) for each container type as counter vectors.

Kuberuntime manager increments the metric/label value when new containers are [started](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L884) or [fail to start](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L890).

If we go this approach operators would monitor HostProcess container usage with the same metrics used see the total number of containers started over time.

Alternatively (or additionally) we can add a gauge metric for `RunningHostProcessContainerCount` which would could mimic the [running_containers](https://github.com/kubernetes/kubernetes/blob/67fa7b278249f12dd29d5b749fe7f973e8f3e0e1/pkg/kubelet/metrics/metrics.go#L443) metric.

Is one approach here better than the other?

Reply:
I prefer the label on existing metric so that Windows host_process isn't different from other containers in those regards.  

I am not a metrics expert but adding labels does increase the number of time series in Prometheus, which could be a consideration but I don't think add one additional label would affect it much?